core/cloudflow-akka/src/main/scala/cloudflow/akkastream/AkkaStreamletLogic.scala

```diff
@@ -132,6 +138,36 @@ abstract class AkkaStreamletLogic(implicit val context: AkkaStreamletContext) ex
   def getSourceWithCommittableContext[T](inlet: CodecInlet[T]): akka.stream.javadsl.SourceWithContext[T, Committable, _] =
     context.sourceWithCommittableContext(inlet).asJava
 
+  /**
+   * This source is designed to function the same as [[sourceWithCommittableContext]]
+   * while also leveraging Akka Kafka Cluster Sharding for stateful streaming.
+   *
+   * This source emits `T` records together with the committable context, thus makes it possible
+   * to commit offset positions to Kafka using `committableSink(outlet: CodecOutlet[T])`.
+   *
+   * It is required to use this source with Akka Cluster.  This source will start up
+   * Akka Cluster Sharding using the supplied `shardEntity` and configure the kafka external
+   * shard strategy to co-locate Kafka partition consumption with Akka Cluster shards.
+   *
+   * @param inlet specifies a [[cloudflow.streamlets.Codec]] that is used to deserialize the records read from the underlying transport.
+   * @param shardEntity is used to specific the settings for the started shard region
+   **/
+  @ApiMayChange
+  def shardedSourceWithCommittableContext[T, M, E](inlet: CodecInlet[T],
```

Comment:
As also mentioned in context I prefer a known mat type over `_`, could you have a look at this for all methods that do this?

Reply:
How does it look now?  Is this better?

Reply:
It is still returning `SourceWithContext[T, CommittableOffset, _]` instead of (for instance) `SourceWithContext[T, CommittableOffset, NotUsed]`

Reply:
maybe my github foo is messed up but when I updated all the signatures to look like this.  When I look at the actual files changed it is updated but for some reason this box isnt 
`  
def shardedSourceWithCommittableContext[T, M, E](inlet: CodecInlet[T],
                                                   shardEntity: Entity[M, E]): SourceWithContext[T, CommittableOffset, Future[NotUsed]] =
    context.shardedSourceWithCommittableContext(inlet, shardEntity)
`

Reply:
Oh yeah now it looks good.